### PR TITLE
Update DNS domain order in preparation for wildcard SSL certs

### PR DESCRIPTION
**Pull request by ajacocks**

### CODE REVIEW

File: content/workshops/pki_workshop/unit_lessons/unit_1.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Unit 1: Installing Directory Server
+
+In this unit you will install [Red Hat Directory Server](https://www.redhat.com/en/technologies/cloud-computing/directory-server "Red Hat Directory Server"). Red Hat Directory Server provides a centralized directory service for an intranet, network, and extranet information. Directory Server integrates with existing systems and acts as a centralized repository for the consolidation of employee, customer, supplier, and partner information. Directory Server can even be extended to manage user profiles, preferences, and authentication. 
+
+Red Hat Directory Server is used as the backend for Certificate System. The setup script can be used interactively but we are going to use an automated install with pre-configured settings.
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_1.md]: Red Hat Directory Server is used as the backend for [the] Certificate System.

File: content/workshops/pki_workshop/unit_lessons/unit_1.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+2. Change to the pki-workshop/unit_labs/rhds directory. This was cloned during the prerequisites.
+
+    `cd pki-workshop/unit_labs/rhds` 
+
+3. Run the setup script. This will automatically configure Directory Server based on the settings in [ds1-setup.ldif](https://gitlab.consulting.redhat.com/pki/pki-workshop/blob/master/unit_labs/rhds/ds1-setup.ldif "ds1-setup.ldif") and [ds1-network.ldif](https://gitlab.consulting.redhat.com/pki/pki-workshop/blob/master/unit_labs/rhds/ds1-network.ldif "ds1-network.ldif").
+
+    `setup-ds.pl --silent --file=ds1-setup.ldif`
+
+4. Ensure Directory Server is running and enabled.
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_1.md]: Ensure [the] Directory Server is running and enabled.

File: content/workshops/pki_workshop/unit_lessons/unit_10.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    `ldapmodify -x -W -h ds1.redhat.example.com -p 389 -D "cn=Directory Manager" -c -f tps-user.ldif`
+
+4. Verify the user was added.
+
+    `ldapsearch -W -h ds1.redhat.example.com -p 389 -D "cn=Directory Manager" -b "ou=People,dc=tps,dc=redhat,dc=example,dc=com"`
+
+## Generate Virtual Token
+
+1. Edit the **phoneHome.xml** file and change the Operation URL from https to http, and ensure the port listed is the unsecure port (by default 8080).
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_10.md]: Edit the **phoneHome.xml** file and change the Operation URL from https to [http. Ensure] the port listed is the [unsecured] port (by default 8080).

File: content/workshops/pki_workshop/unit_lessons/unit_10.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    `vim /etc/pki/ca1/tps/phoneHome.xml`
+
+2. Restart the CA.
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_10.md]: What is the CA? We should define it in this new lesson.

File: content/workshops/pki_workshop/unit_lessons/unit_11.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
+# Unit 11: Certificate profiles
+
+In this unit you will go over the caUserCert profile. This certificate profile is for enrolling user certificates.
+
+The CA profiles are located here:
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_11.md]: CA should be defined on first use in each unit.

File: content/workshops/pki_workshop/unit_lessons/unit_11.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+    /var/lib/pki/ca1/ca/profiles/ca
+
+The [caUserCert.cfg](https://gitlab.consulting.redhat.com/pki/pki-workshop/blob/master/unit_labs/rhcs/caUserCert.cfg) profile is located in the rhcs labs folder.
+
+## Example caUserCert Profile
+
+The first part of a certificate profile is the description. This shows the name, long description, whether it is enabled, and who enabled it.
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_11.md]: The fields you will see in a CA User Profile are as follows:

- desc is a long description of the user certificate profile
- visible (true/false) controls whether the certificate will be visible to users
- enable, when selected (true), will allow the certificate to be used
- enableBy identifies the user who enabled the certificate
- name is the short name used for the certificate that will be viewable by users

File: content/workshops/pki_workshop/unit_lessons/unit_1.md (hunk 1 above)
Comment (ajacocks) [reply content/workshops/pki_workshop/unit_lessons/unit_1.md]: Fixed

File: content/workshops/pki_workshop/unit_lessons/unit_1.md (hunk 2 above)
Comment (ajacocks) [reply content/workshops/pki_workshop/unit_lessons/unit_1.md]: Fixed

File: content/workshops/pki_workshop/unit_lessons/unit_10.md (hunk 3 above)
Comment (ajacocks) [reply content/workshops/pki_workshop/unit_lessons/unit_10.md]: Fixed

File: content/workshops/pki_workshop/unit_lessons/unit_10.md (hunk 4 above)
Comment (ajacocks) [reply content/workshops/pki_workshop/unit_lessons/unit_10.md]: Fixed

File: content/workshops/pki_workshop/unit_lessons/unit_11.md (hunk 5 above)
Comment (ajacocks) [reply content/workshops/pki_workshop/unit_lessons/unit_11.md]: Fixed

File: content/workshops/pki_workshop/unit_lessons/unit_11.md (hunk 6 above)
Comment (ajacocks) [reply content/workshops/pki_workshop/unit_lessons/unit_11.md]: Fixed

File: content/workshops/pki_workshop/unit_lessons/unit_11.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+***
+
+The missing auth.instance_id= entry in this profile means that with this profile, authentication is not needed to submit the enrollment request.
+
+    auth.class_id=
+
+***
+
+Next, the profile lists all of the required inputs for the profile. For the **caUserCert** profile, this defines the keys to generate, the fields to use in the subject name, and the fields to use for the person submitting the certificate.
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_11.md]: Next, in our case of the **caUserCert** profile, we view the required profile inputs. (input.list)
input.i1.class_id=keyGenInputImpl - defines the keys to generate
input.i2.class_id=subjectNameInputImpl - defines the keys to use in the subject name
input.i3.class_id=submitterInfoInputImpl - defines the fields to use for the person submitting the certificate

**Review by matmille — APPROVED** [review]

File: content/workshops/pki_workshop/unit_lessons/unit_10.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+The admin certificate needs to be in your browser. This was completed during [Unit 4](https://gitlab.consulting.redhat.com/pki/pki-workshop/blob/master/unit_lessons/unit_4.md).
+
+* Open the TPS webpage.
+
+    `https://ca1.redhat.example.com:8443/tps/`
+
+* Click the **Log In** button.
+
+* Go through the different tabs to become familiar with the interface. It will be mostly blank until we create users and tokens.
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_10.md]: Go through each tab (or) Go through the tabs ("different" is not required)

File: content/workshops/pki_workshop/unit_lessons/unit_12.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
+# Unit 12: Troubleshooting
+
+In this unit you will troubleshoot common issues with maintaining a Certificate System. Each section has a shell script to run that will "break" the system. A hint will be offered to help you hone your troubleshooting skills.
+
+## Instructions
+
+To solve, the CA should be accessible in a browser without errors in status output or logs.
+
+### Run script to break CA (1st lab)
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_12.md]: ### Run script to break certificate authority (CA) (1st lab)

File: content/workshops/pki_workshop/unit_lessons/unit_12.md [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+3. Troubleshoot.
+
+### Solution
+
+<details>
+<summary>Click to expand solution.</summary>
+
+<ol>
+  <li>The webpage is shows a java error but no real clues.</li>
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_12.md]: Java

File: content/workshops/pki_workshop/unit_lessons/unit_3.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+            ==========================================================================
+
+
+3. Check status of TKS.
+
+    `systemctl status -l pki-tomcatd@ca1.service`
+
+## TPS
+A Token Processing System (TPS). The TPS interacts directly with external tokens, like smart cards, and manages the keys and certificates on those tokens through a local client, the Enterprise Security Client (ESC). The ESC contacts the TPS when there is a token operation, and the TPS interacts with the CA, KRA, or TKS, as required, then send the information back to the token by way of the Enterprise Security Client. 
+
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_3.md]: then [sends] the information back to the token by way of the Enterprise Security Client.

File: content/workshops/pki_workshop/unit_lessons/unit_4.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
+# Unit 4: Accessing the Web Interface
+
+In this unit you will access the web interface of each subsystem. Each Certificate System instance is a Tomcat server instance and each Certificate System subsystem (such as CA or KRA) is deployed as a web application in Tomcat.
+
+All subsystems share a common administrative interface and have an agent interface that allows for agents to perform the tasks assigned to them. A CA Subsystem has an end-entity interface that allows end-entities to enroll in the PKI. An OCSP Responder subsystem has an end-entity interface allowing end-entities and applications to check for current certificate revocation status. Finally, a TPS has an operator interface.
+
```
Comment (Tekwriter1221) [inline content/workshops/pki_workshop/unit_lessons/unit_4.md]: All subsystems share a common administrative interface and have an agent interface that [enables] agents to perform the tasks assigned to them. A CA [subsystem] has an end-entity interface that [enables] end-entities to enroll in the PKI. An OCSP Responder subsystem has an end-entity interface, [which enables] end-entities and applications to check for current certificate revocation status. Finally, a TPS has an operator interface.

**Review by Tekwriter1221 — COMMENTED** [review]
Hey Alex, I put a bunch of comments in the whole set, for your review....

**Pull request merged**